### ISSUE TITLE
Feature Request: Support for Amazon Aurora Backtrack

### ISSUE THREAD

**Issue by kylegoch** [terraform-providers/terraform-provider-aws#4498]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Recently added support for "Backtrack" In Aurora for MySQL clusters. 

This feature can only be applied to new clusters or clusters restored from a snapshot. It can not be applied to existing clusters. TF would need to error or prevent that.

### New or Affected Resource(s)

* aws_rds_cluster (resource)

### Potential Terraform Configuration

```hcl
resource "aws_rds_cluster" "default" {
  cluster_identifier      = "aurora-cluster-demo"
  engine                  = "aurora-mysql"
  availability_zones      = ["us-west-2a", "us-west-2b", "us-west-2c"]
  database_name           = "mydb"
  master_username         = "foo"
  master_password         = "bar"
  backtrack_window        = 24
  backup_retention_period = 5
  preferred_backup_window = "07:00-09:00"
}
```

### References

* https://aws.amazon.com/blogs/aws/amazon-aurora-backtrack-turn-back-time/
* https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBCluster.html
* https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_BacktrackDBCluster.html
* https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBClusterBacktracks.html

**Comment by bflad** [comment]
This is pretty exciting!

A `backtrack_window` argument on the `aws_rds_cluster` resource sounds like a reasonable implementation (just note that the value is in seconds in the API 😉 ). It'll need to be handled in all these operations:
* `CreateDBCluster`
* `DescribeDBCluster`
* `RestoreDBClusterFromS3`
* `RestoreDBClusterFromSnapshot`
* `ModifyDBCluster`

Given the other references above to `BacktrackDBCluser` in the API documentation, I am personally not sure managing actual backtracks via a new `aws_rds_cluster_backtrack` resource makes too much sense in Terraform as this seems like a very time-dependent/one-off operation. I would suggest creating a separate issue if there really is strong demand for that separate new resource. (Also, would be very curious about the use cases there!)

**Comment by bflad** [comment]
Pull request submitted: #4524

**Comment by bflad** [comment]
The `backtrack_window` argument has been added to the `aws_rds_cluster` resource and will release with v1.19.0 of the AWS provider later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.19.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1190-may-16-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
